app/os_windows.go

```diff
@@ -954,4 +987,95 @@ func configForDPI(dpi int) unit.Metric {
 	}
 }
 
+// schemesURI is a list of schemes, comma separated, that must be
+// defined using -X compiler ldflag, that used in gogio.
+var schemesURI string
+var schemesURIList []string
+
+// startupURI is the URI that started the app, if any.
+var startupURI string
+
+func init() {
+	if schemesURI == "" {
+		return
+	}
+
+	schemesURIList = strings.Split(schemesURI, ",")
+
+	if len(os.Args) >= 2 {
+		startupURI = os.Args[1]
+	}
+
+	code, err := windows.RegisterWindowMessage(schemesURI)
+	if err != nil {
+		return
+	}
+	windows.GIO_OPEN_URL = code
+
+	/*
+		On Windows, launching the app using a URI will start a new instance of the app,
+		a new window. That behavior doesn't align with iOS/Android/macOS, where the deeplink
+		sends the event to the running app (if any).
+
+		In order to align the behavior, we use a global mutex to check if the app is already
+		running. If it is, we send the deeplink to the running app and exit the new instance.
+
+		That should happen on init to prevent the user from seeing the new window.
+	*/
+	if !createURISharedMemory() {
+		if startupURI != "" {
+			broadcastURI(startupURI)
+		}
+		os.Exit(0)
+		return
+	}
+
+	for _, scheme := range schemesURIList {
+		go windows.RegisterScheme(scheme)
+	}
+}
+
+func createURISharedMemory() (ok bool) {
+	if _, err := windows.CreateFileMapping(windows.PAGE_READWRITE, schemesURI, 1024*8); err != nil {
+		return false
+	}
+	return true
+}
+
+func readURI(wParam uintptr) string {
+	mmap, err := windows.OpenFileMapping(windows.FILE_MAP_READ, schemesURI)
+	if err != nil {
+		return ""
+	}
+	defer windows.CloseFileMapping(mmap)
+
+	src, err := windows.MapViewOfFile(mmap, windows.FILE_MAP_READ, 1024*8)
+	if err != nil {
+		return ""
+	}
+	defer windows.UnmapViewOfFile(src)
+
+	if wParam >= uintptr(len(src)) {
+		return ""
+	}
+	return string(src[:wParam])
+}
+
+func broadcastURI(uri string) {
+	mmap, err := windows.OpenFileMapping(windows.FILE_MAP_WRITE, schemesURI)
+	if err != nil {
+		return
+	}
+	defer windows.CloseFileMapping(mmap)
+
+	dst, err := windows.MapViewOfFile(mmap, windows.FILE_MAP_WRITE, 1024*8)
+	if err != nil {
+		return
+	}
```

Comment:
I tried to use `WM_COPYDATA` but it requires to know the specific HWND of the receiver. Also, in both cases it will need to have mutex, which `named-mmap` is also used for. 

The receiver creates the mmap using the scheme as the name of the mmap. The sender checks if the mmap already exists, if exists writes the URL into the mmap and then sends a event, that event don't contain any pointer or mmap.

Reply:
> I tried to use WM_COPYDATA but it requires to know the specific HWND of the receiver.

If Windows opens a particular Gio binary, you should be able to locate any other running instances of that binary and only target them with WM_COPYDATA. For example, by the window class name or window name/id.

> Also, in both cases it will need to have mutex, which named-mmap is also used for.

See above, I don't think you need a mutex if you just target windows belonging to the binary being opened.

Reply:
I found one issue related to UNC. I tried to search by path, then send COPYDATA to that executable. 

For instance, it's possible to have one `X:/demo.exe` and `//192.168.0.1/demo.exe`, the first one is one "map" to the second one. Here is the issue, because I can't use `os.Executable() == anotherExePath`.

Because of that edge-case, I'm comparing the `filepath.Base`. However, `Base` may cause collision (for instance your/app.exe is the same of randompath/app.exe). 

I think another alternative is to use `-appid` (?) from `gogio` and also checks the `WindowClassName` (which is currently hardcoded as "Gio"), not just the `filepath.Base`.

Reply:
I notice that `appID` is set in `internal/log`, and used only for Android. I create a new variable, in `os_windows.go`, but seems that both have the same purpose.

Maybe we should create some `app/meta/meta.go` which contains `AppID`, `Version`, `URLSchemes` (...)? That makes easier to set variables (without gogio) and also re-use it anywhere.

I will leave as duplicated, and in some next patch create such package.